src/hlsBinaries.ts

```diff
@@ -243,14 +243,14 @@ async function getLatestReleaseMetadata(context: ExtensionContext, storagePath:
   try {
     const releaseInfo = await httpsGetSilently(opts);
     const latestInfoParsed =
-      validate.parseAndValidate(releaseInfo, githubReleaseApiValidator).find((x) => !x.prerelease) || null;
+      validate.parseAndValidate(releaseInfo, githubReleaseApiValidator).filter((x) => !x.prerelease) || null;
 
     if (updateBehaviour === 'prompt') {
       const cachedInfoParsed = await readCachedReleaseData();
 
       if (
         latestInfoParsed !== null &&
-        (cachedInfoParsed === null || latestInfoParsed.tag_name !== cachedInfoParsed.tag_name)
+        (cachedInfoParsed === null || latestInfoParsed[0].tag_name !== cachedInfoParsed[0].tag_name)
       ) {
```

Comment:
Assuming [0] always exists. Add extra check if this doesn't hold.

Reply:
that check would be great, could be it triggered by a failing http request?